contracts/ICO_controller.sol

```diff
@@ -215,24 +215,30 @@ contract ICO_controller is Ownable, TransferableInterface {
             // sends token for support program
             bool success = token.transfer(incentiveProgram, INCENTIVE_PROGRAM_SUPPORT);
             assert(success==true);
+            // send 50% of ico eth to contract onwer
+            escrowIco.transfer(this.balance.div(2));
+            // send other 50% to multisig holder address
+            holder.transfer(this.balance);
+        }
+        crowdsaleFinished = true;
+
+    }
+
+    function finishCrowdsaleBurnUnused() external onlyOwner {
+        require(crowdsaleFinished);
+        if (totalSold >= SOFTCAP) {
             // burn some unspent reward tokens
             token.burn(MAX_DEV_REWARD.sub(totalDevReward));
             // burn after airdrop left tokens
             if (totalAirdropAdrresses != 0) {
                uint256 airdropToBurn = AIRDROP_SUPPLY.sub(AIRDROP_SUPPLY.div(totalAirdropAdrresses).mul(totalAirdropAdrresses));
                 if (airdropToBurn != 0){
                     token.burn(airdropToBurn);
-                } 
+                }
```

Comment:
Please also check before `token.burn(MAX_DEV_REWARD.sub(totalDevReward))` that we are trying to burn > 0 tokens and that we have more tokens on the balance than we are trying to burn.

Also when burning airdrop leftovers - check that we have enough tokens on the balance.